sfge/src/test/java/com/salesforce/graph/source/supplier/AnnotationSourceSupplierTest.java

```diff
@@ -0,0 +1,54 @@
+package com.salesforce.graph.source.supplier;
+
+import com.salesforce.collections.CollectionUtil;
+import com.salesforce.graph.Schema;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/**
+ * Annotation suppliers generally work the same, so they can all be put into a single file with
+ * parameterized tests.
+ */
+public class AnnotationSourceSupplierTest extends BaseSourceSupplierTest {
+    // spotless:off
+    private static final String TEMPLATE =
+        "public class Foo {\n"
+        // Add a method with a configurable annotation.
+      + "    @%s\n"
+      + "    public boolean annotatedMethod() {\n"
+      + "        return true;\n"
+      + "    }\n"
+      + "    \n"
+        // Add a method without an annotation.
+      + "    public boolean nonAnnotatedMethod() {\n"
+      + "        return true;\n"
+      + "    }\n"
+      + "}\n";
+    // spotless:on
+
+    /**
+     * Suppliers that check for a given annotation should return methods that are appropriately
+     * annotated.
+     */
+    @MethodSource("supplierLoadsAnnotatedMethods_params")
+    @ParameterizedTest(name = "{displayName}: Annotation @{0}")
+    public void supplierLoadsAnnotatedMethods(
```

Comment:
Equivalent to `PathEntryPointUtilTest#getPathEntryPoints_includesAnnotatedMethods` and `PathEntryPointUtilTest#testGetMethodsWithAnnotation`.